### DISCUSSION TITLE
Is it possible to use environment variables within `mackup.cfg`?

### DISCUSSION THREAD

**Discussion by noraworld** [lra/mackup#1972] [Q&A]
I would like to use environment variables within `mackup.cfg`, but I don't know how to achieve this. The example of the usage is something like this:

```
[storage]
engine = file_system
path = workspace/mackup
directory = $OSTYPE
```

I want to separate directories by OSs.

I read https://github.com/lra/mackup/issues/618, but I think this only mentions about XDG.

**Comment by joshmedeski** [comment]
I don't believe this is supported at this time. I recommend using a different `~/.mackup.cfg` file for each computer. Just be careful when running the `mackup restore` command and make sure you don't overwrite your local `~/.mackup.cfg`.

Hope that helps.

**Comment by noraworld** [comment]
@joshmedeski Thank you for your reply.

Then I will use it by separating files by each OSs for now. But I think it is more useful for us to be able to use environment variables or other alternatives. Is there a plan to implement this?

**Comment by joshmedeski** [comment]
Issues have been brought up multiple times over the year. If someone wants to create a proposal that keeps mackup simple, we are open to it.

The main concern is creating a solution that makes mackup less beginner-friendly or complicates the setup.

**Comment by noraworld** [comment]
I see. Unfortunately, I don't have enough time and knowledge about it, but I will be waiting for a new solution coming.

Thank you for letting me know. 🙏

**Comment by lra** [comment]
You can maybe workaround this by adding links to the same folder on different OSes?

**Comment by Wxh16144** [comment]
I have a similar need

For example, I want to overwrite the zsh configuration, and I want to back up my newly added zsh shell using `mackup`

The file looks like this:

<img width="1121" alt="image" src="https://user-images.githubusercontent.com/32004925/210359351-e7f3fcb5-198b-4626-be27-fa10b4947cec.png">
